pkgs/development/libraries/rocsolver/default.nix

```diff
@@ -0,0 +1,104 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, writeScript
+, cmake
+, rocm-cmake
+, rocm-runtime
+, rocm-device-libs
+, rocm-comgr
+, rocblas
+, hip
+, fmt
+, gtest ? null
+, gbenchmark ? null
+, buildTests ? false
+, buildBenchmarks ? false
+, gpuTargets ? null # gpuTargets = [ "gfx803" "gfx900" "gfx906:xnack-" ]
+}:
+
+assert buildTests -> gtest != null;
+assert buildBenchmarks -> gbenchmark != null;
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "rocsolver";
+  repoVersion = "3.19.0";
+  rocmVersion = "5.3.3";
+  version = "${finalAttrs.repoVersion}-${finalAttrs.rocmVersion}";
```

Comment:
```suggestion
  version = "${finalAttrs.rocmVersion}";
```

Reply:
I don't think that's currently a good idea.
``3.19.0`` may not be compatible with programs using X.X.X.
``repoVersion`` allows us to know in the nixpkgs side of things.
The ``version`` being a combo causes the output to have both versions, so the user-side can adjust if needed.

Reply:
It is also possible for the repo to update it's source without updating the rocm version.

Reply:
> It is also possible for the repo to update it's source without updating the rocm version.

I think it’s fine if we only package rocm versions in nixpkgs and ignore the repoVersion.
If there are changes to the repoVersion, we probably shouldn’t package them anyway but wait for the next rocm release that contains them so that the rocm version of all packages is in sync.

Reply:
I see what you mean, but again, compatibility cannot be guaranteed if we do that.
``miopen`` may depend on `2.45.0`` of ``rocblas``, but may not work with ``2.44.0``, despite them both being the same rocm version.
It's that sort of issue.

Reply:
How would that happen?
The rocm-\<ver> tag can only point to a single commit and that commit of `rocblas` and that commit is either 2.45.0 or 2.44.0 but not both.

As far as I understand, a rocm release packages all of the packages at a specific commit – fixed by the rocm-\<ver> tag – and this rocm release is tested at AMD and then released as packages and source on GitHub.
So, the tested configuration is using the rocm-\<ver> tag of all repos.
If we package rocm-5.3.3 of `miopen`, we should also package rocm-5.3.3 of `rocblas`, no matter what the repoVersion is.

Reply:
It does seem to be kind of chicken and the egg to me.
Whatever, let's just remove ``repoVersion`` and see what happens.